### PR TITLE
Support yarn licenses command in yarn v2+

### PR DESCRIPTION
**Pull request by julia-pu**

### CODE REVIEW

File: lib/license_finder/package_managers/yarn.rb [hunk 1]
```diff
@@ -2,16 +2,20 @@
 
 module LicenseFinder
   class Yarn < PackageManager
-    SHELL_COMMAND = 'yarn licenses list --no-progress --json'
```
Comment (Pivotal-Jeff-Jun) [inline lib/license_finder/package_managers/yarn.rb:5]: is `--no-progress` a deprecated flag? or we don't need it anymore?

File: lib/license_finder/package_managers/yarn.rb [hunk 2]
```diff
@@ -2,16 +2,20 @@
...
 
     def current_packages
+      #keeping the use of yarn1_production_flag here because the license plugin supports same commands as licenses in yarn v1
       cmd = "#{Yarn::SHELL_COMMAND}#{yarn1_production_flag}"
-      suffix = " --cwd #{project_path}" unless project_path.nil?
-      cmd += suffix unless suffix.nil?
+      unless yarn2_project?
+        cmd += " --no-progress"
+        suffix = " --cwd #{project_path}" unless project_path.nil?
+        cmd += suffix unless suffix.nil?
```
Comment (Pivotal-Jeff-Jun) [inline lib/license_finder/package_managers/yarn.rb:17]: could we refactor these two lines into one?
```ruby
cmd += " --cwd #{project_path}" unless project_path.nil?
```

File: lib/license_finder/package_managers/yarn.rb (hunk 1 above)
Comment (Pivotal-Jeff-Jun) [reply lib/license_finder/package_managers/yarn.rb:5]: or is that this flag is only in yarn 2 and not 1

File: lib/license_finder/package_managers/yarn.rb [hunk 3]
```diff
@@ -83,6 +92,16 @@ def yarn2_project?
       end
     end
 
+    def yarn3_project?
```
Comment (xtreme-shane-lattanzio) [inline lib/license_finder/package_managers/yarn.rb:95]: This should be refactored to just return the project type. Duplicating the same code and then checking for a number isnt great

File: lib/license_finder/package_managers/yarn.rb [hunk 4]
```diff
@@ -2,16 +2,20 @@
...
   class Yarn < PackageManager
-    SHELL_COMMAND = 'yarn licenses list --no-progress --json'
+    SHELL_COMMAND = 'yarn licenses list --json'
 
     def possible_package_paths
       [project_path.join('yarn.lock')]
     end
 
     def current_packages
+      #keeping the use of yarn1_production_flag here because the license plugin supports same commands as licenses in yarn v1
```
Comment (xtreme-shane-lattanzio) [inline lib/license_finder/package_managers/yarn.rb:12]: Should this flag be renamed then?

File: lib/license_finder/package_managers/yarn.rb [hunk 5]
```diff
@@ -66,7 +70,12 @@ def prepare_command
     private
 
     def yarn2_prepare_command
```
Comment (xtreme-shane-lattanzio) [inline lib/license_finder/package_managers/yarn.rb:72]: The prepare_command on line 62 only checks for yarn1 or 2. What is it supposed to be if it is yarn 3? I also don't like how this yarn2 prepare command is for yarn 3 as well. This file needs to be refactored

**Review by xtreme-shane-lattanzio — CHANGES_REQUESTED** [review]
This should be cleaned up. Let me know if you wanna chat

File: lib/license_finder/package_managers/yarn.rb (hunk 1 above)
Comment (julia-pu) [reply lib/license_finder/package_managers/yarn.rb:5]: that flag is only supported in yarn 1, so I'm removing it from the base shell command and appending it only if it's yarn v1

File: lib/license_finder/package_managers/yarn.rb (hunk 5 above)
Comment (julia-pu) [reply lib/license_finder/package_managers/yarn.rb:72]: the check for yarn2 actually is checking for any versions greater or equal to yarn2, so it includes yarn 3. I was originally trying to change as little things as possible, but I agree it is confusing. I'll do some refactoring

Comment (julia-pu) [reply lib/license_finder/package_managers/yarn.rb]: made the refactor

File: lib/license_finder/package_managers/yarn.rb (hunk 4 above)
Comment (julia-pu) [reply lib/license_finder/package_managers/yarn.rb]: changed in the refactor

File: lib/license_finder/package_managers/yarn.rb (hunk 3 above)
Comment (julia-pu) [reply lib/license_finder/package_managers/yarn.rb]: refactored

File: lib/license_finder/package_managers/yarn.rb (hunk 2 above)
Comment (julia-pu) [reply lib/license_finder/package_managers/yarn.rb]: refactored

**Comment by julia-pu** [comment]
Refactored, please take a look again!

**Review by xtreme-shane-lattanzio — APPROVED** [review]
Much better. Thank you!!!

**Review by xtreme-debbie-chen — APPROVED** [review]

**Pull request merged**